Whitelist plugins per organization on Cloud

Twixes opened the pull request:
## Changes

This allows whitelisted organizations (with a comma-separated org IDs env var `PLUGINS_CLOUD_WHITELISTED_ORG_IDS`) to manage plugins on Cloud.
The `Plugin` model gets a new field `organization`, which is an `Organization` foreign key, ~~nullable (grandfathered in `Plugin` rows have it null)~~ set to the first Organization present in the PostHog instance.
Also reworks the plugins API from `/api/plugin/:plugin-id/` to `/api/organizations/:organization-id/plugins/:plugin-id/` using `StructuredViewSetMixin`. This means that the endpoint's queryset gets autofiltered to only return plugins with the specific `organization` foreign key value.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

Twixes committed:
Fix unix-dgram build

Twixes committed:
Add per-organization whitelisting of plugins API on Cloud

Twixes committed:
Add migration

Twixes committed:
Update can_install_plugins_via_api calls in serializer

Twixes committed:
Remove unused type: ignore

Twixes committed:
Update PluginSerializer for organization_id

Twixes committed:
Make Plugin.organization nullable

mariusandra commented:
Hey! I didn't run the code yet, but reading through it some things to add:
- Tests seem to be failing
- `can_configure_plugins_via_api` will return false always in the cloud, so nobody can configure plugins, only install...
- I see no filtering when fetching plugins to limit the organisation. Is this built in to the viewset somehow?
- Tests to see that the API only returns plugins belonging to your org (and you can't create/update those of other orgs)
- I'd add something that sets the first organization in the migration, instead of leaving it at null

Twixes committed:
Fix can_install_plugins_via_api

Twixes committed:
Update pluginsLogic.ts

Twixes committed:
Adjust can_configure_plugins_via_api for organization

Twixes committed:
Don't include organization_id in serializer

Twixes committed:
Set default Plugin.organization to first Organization in instance

Twixes committed:
Fix PluginConfigSerializer.create

Twixes committed:
Test that other orgs' plugins can't be accessed (as if they didn't exist, 404)

Twixes commented:
Ad 1. Fixed.
Ad 2. Made `can_configure_plugins_via_api` and `can_configure_plugins_via_api` work the same through `guard_cloud(organization_or_id)`.
Ad 3. Yep, `StructuredViewSetMixin` does autofiltering (mentioned in original post up top).
Ad 4. Added a test. There isn't really much to test here, as it's just `StructuredViewSetMixin` slapping a `.filter()` on the queryset, so to other orgs/projects it's simply as if the Plugin/PluginConfig instance didn't exist.
Ad 5. Fair point, done.

mariusandra committed:
fix typo

mariusandra reviewed (approved):
LGTM!
